core/src/main/java/com/picone/core/data/repository/RestaurantDaoImpl.java

```diff
@@ -25,4 +48,60 @@ public RestaurantDaoImpl(FirebaseDatabase database){
     public Restaurant getRestaurant(int position) {
         return null;
     }
+
+    @Override
+    public Observable<List<Restaurant>> getRestaurantForName(String restaurantName) {
+        Query query = restaurantsDataBaseReference.orderByChild("name").equalTo(restaurantName);
+        return RxFirebaseDatabase.observeValueEvent(query, DataSnapshotMapper.listOf(Restaurant.class)).toObservable();
+    }
+
+    @Override
+    public Observable<List<Restaurant>> getRestaurantForKey(String restaurantKey){
+        Query query = restaurantsDataBaseReference.orderByChild("key").equalTo(restaurantKey);
+        return RxFirebaseDatabase.observeValueEvent(query, DataSnapshotMapper.listOf(Restaurant.class)).toObservable();
+    }
+
+    @Override
+    public Completable addRestaurant(Restaurant restaurant) {
+        restaurant.setKey(restaurantsDataBaseReference.child(restaurant.getName()).push().getKey());
+        return RxFirebaseDatabase.setValue(restaurantsDataBaseReference
+                .child(restaurant.getName()), restaurant);
+    }
+
+    @Override
+    public Completable updateUserChosenRestaurant(String currentUserEmail, UserDailySchedule userDailySchedule) {
+        Query query = database.getReference().child("users").orderByChild("email").equalTo(currentUserEmail);
+
+        return Completable.create(emitter ->
+                query.addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(@NonNull DataSnapshot snapshot) {
+                        for (DataSnapshot dataSnapshot : snapshot.getChildren()) {
+                            dataSnapshot.getRef().child("userDailySchedule").setValue(userDailySchedule)
+                                    .addOnSuccessListener(
+                                            aVoid -> emitter.onComplete())
+                                    .addOnFailureListener(
+                                            e -> {
+                                                if (!emitter.isDisposed())
+                                                    emitter.onError(e);
+                                            });
+                        }
+                    }
+
+                    @Override
+                    public void onCancelled(@NonNull DatabaseError error) {
```

Comment:
emitter.onError ?

Reply:
I have made the same way Rx2Firebase done, i have show how they write their setValue method and adapt it to my usage.
So i think it is to throw an error on writing dailyShedule.

Reply:
Can't refactor this method without refactoring addUser.

. If i push firebase create a new user.
. If i don't push, firebase remove all users and write new user in root.

To resolve this i have to not push user when added, but write under a node like "name" or "uid".

Do i have to make changes?

Reply:
So to do that way i have to add a parameter "key" in User model in order to get the key.

In firebase the getKey method could only be access with the push method, so i could only do this when i push user.